From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: tabrisnet <tabris@tabris.net>
Date: Mon, 9 Feb 2026 20:13:04 +0000
Subject: Patching kernel filogic files

connects phy to NVMe port and to one of the mPCIe slots

Signed-off-by: tabrisnet <tabris@tabris.net>
---
 arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi | 27 ++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
index 358ac1df35dd..aec6a68b756c 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
@@ -342,25 +342,52 @@ i2c_wifi: i2c@3 {
 };
 
 /* mPCIe SIM2 */
 &pcie0 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie0_pins>; /* ensure pinctrl matches pcie0! */
+
+	/* Connect mPCIe lane to the XS-PHY */
+	phys = <&xsphy 0>; /* Lane 0 */
+	phy-names = "pcie-phy";
+
+	/* The NVMe Reset line on R4 is typically pin 8 */
+	reset-gpios = <&pio 8 GPIO_ACTIVE_LOW>;
 };
 
 /* mPCIe SIM3 */
 &pcie1 {
 	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie1_pins>; /* ensure pinctrl matches pcie1! */
+
+	/* there are only 2 xs-phy lanes available
+	   so have to use t-phy for this port */
 };
 
 /* M.2 key-B SIM1 */
 &pcie2 {
 	status = "okay";
 };
 
 /* M.2 key-M SSD */
 &pcie3 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie3_pins>; /* ensure pinctrl matches pcie3! */
+
+	/* Connect NVMe lanes to the XS-PHY */
+	phys = <&xsphy 1>; /* Lane 1 */
+	phy-names = "pcie-phy";
+
+	/* The NVMe Reset line on R4 is typically pin 7 */
+	reset-gpios = <&pio 7 GPIO_ACTIVE_LOW>;
+
+	/* Ensure the 3.3V rail is active */
+	vpcie3v3-supply = <&reg_3p3v>;
 };
 
 &pio {
 	i2c0_pins: i2c0-g0-pins {
 		mux {
-- 
Created with Armbian build tools https://github.com/armbian/build

